source/opt/def_use_manager.cpp

```diff
@@ -103,24 +105,59 @@ bool DefUseManager::ReplaceAllUsesWith(uint32_t before, uint32_t after) {
 
 void DefUseManager::ClearDef(uint32_t def_id) {
   if (id_to_def_.count(def_id) == 0) return;
+  ClearInst(id_to_def_.at(def_id));
+}
 
-  // Go through all ids used by this instruction, remove this instruction's
-  // uses of them.
-  for (const auto use_id : result_id_to_used_ids_[def_id]) {
-    if (id_to_uses_.count(use_id) == 0) continue;
-    auto& uses = id_to_uses_[use_id];
-    for (auto it = uses.begin(); it != uses.end();) {
-      if (it->inst->result_id() == def_id) {
-        it = uses.erase(it);
-      } else {
-        ++it;
+void DefUseManager::ClearInst(ir::Instruction* inst) {
+  // Do nothing if the instruction is a nullptr or it has not been analyzed
+  // before.
+  if (!inst || analyzed_insts_.count(inst) == 0) return;
+
+  // Go through all ids, except the result id, used by this instruction, remove
+  // this instruction's uses of those ids.
+  for (uint32_t i = 0; i < inst->NumOperands(); i++) {
+    if (ShouldRecord(inst->GetOperand(i))) {
+      uint32_t operand_id = inst->GetSingleWordOperand(i);
+      // Skip if the operand id is not recorded.
+      if (id_to_uses_.count(operand_id) == 0) {
```

Comment:
Fixed in 4d7c019.
Actually the iteration of the operands are also bundled in a private function.